### PR TITLE
Implement a dummy GetBucketACL for legacy reasons

### LINKED ISSUE #5706: Ignore unsupported object ACL header?
I’m trying to use Minio to emulate S3 for an application I’m developing. My app uses a library to upload files/put objects into S3, and I don’t have much control over it. Unfortunately it applies an object-level ACL to every file uploaded, which I know Minio doesn’t support (#4496).

My issue is that Minio returns a 501 error, `A header you provided implies functionality that is not implemented`, on such attempted uploads. Would it be possible for Minio to just ignore this unsupported header? Possibly via configuration?

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
GetBucketACL call simply returns empty for all GET
in ACL requests, the primary purpose of this PR
is simply to provide legacy API support for
legacy applications.

## Motivation and Context
Fixes #5706

## How Has This Been Tested?
Manually and using legacy applications

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by balamurugana** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :x: [more...](#d39b3c07-479f-4952-bf86-53900fc4edf2) |
<a name="d39b3c07-479f-4952-bf86-53900fc4edf2"></a>
### 5906-fc41c3b/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT: minikube:30452
ACCESS_KEY:      minio
SECRET_KEY:      ***REDACTED***
ENABLE_HTTPS:    0
SERVER_REGION:   us-east-1
MINT_DATA_DIR:   /mint/data
MINT_MODE:       core

To get logs, run 'docker cp fad48500b96b:/mint/log /tmp/mint-logs'
(1/13) Running aws-sdk-go tests ... done in 4 seconds
(2/13) Running aws-sdk-java tests ... done in 19 seconds
(3/13) Running aws-sdk-php tests ... done in 54 seconds
(4/13) Running aws-sdk-ruby tests ... done in 49 seconds
(5/13) Running awscli tests ... done in 4 minutes and 11 seconds
(6/13) Running mc tests ... done in 3 minutes and 36 seconds
(7/13) Running minio-dotnet tests ... done in 1 minutes and 0 seconds
(8/13) Running minio-go tests ... done in 13 seconds
(9/13) Running minio-java tests ... done in 52 seconds
(10/13) Running minio-js tests ... FAILED in 53 seconds
{
  "name": "minio-js",
  "function": "listenBucketNotification(bucketName, prefix, suffix, events)",
  "args": "bucketName:minio-js-test-6d537f39-b4d6-4eb5-9329-752b61b2b39b, events: s3:ObjectCreated:*",
  "duration": 8344,
  "status": "FAIL",
  "error": "Error: notification not received at client.removeObject.err (test/functional-tests.js:1034:43) at ClientRequest.<anonymous> (node_modules/minio/dist/main/minio.js:420:9) at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:478:21) at HTTPParser.parserOnHeadersComplete (_http_common.js:99:23) at Socket.socketOnData (_http_client.js:367:20) at readableAddChunk (_stream_readable.js:176:18) at Socket.Readable.push (_stream_readable.js:134:10) at TCP.onread (net.js:559:20)"
}

Executed 9 out of 13 tests successfully.
```

**Review by nitisht — APPROVED** [review]

**Comment by balamurugana** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |

File: cmd/generic-handlers.go [hunk 1]
```diff
@@ -470,6 +470,12 @@ func (h resourceHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	// If bucketName is present and not objectName check for bucket level resource queries.
 	if bucketName != "" && objectName == "" {
+		// enable GetBucketACL dummy call specifically.
+		_, ok := r.URL.Query()["acl"]
+		if r.Method == http.MethodGet && ok {
+			h.handler.ServeHTTP(w, r)
+			return
+		}
```
Comment (krishnasrinivas) [inline cmd/generic-handlers.go]: suggestion: ignoreNotImplementedBucketResources() and ignoreNotImplementedObjectResources() is a better place to abstract this detail.

i.e
```
@@ -420,6 +420,9 @@ func setIgnoreResourcesHandler(h http.Handler) http.Handler {
 // Checks requests for not implemented Bucket resources
 func ignoreNotImplementedBucketResources(req *http.Request) bool {
        for name := range req.URL.Query() {
+               if name == "acl" && req.Method == http.MethodGet {
+                       return false
+               }
                if notimplementedBucketResourceNames[name] {
                        return true
                }
@@ -430,6 +433,9 @@ func ignoreNotImplementedBucketResources(req *http.Request) bool {
 // Checks requests for not implemented Object resources
 func ignoreNotImplementedObjectResources(req *http.Request) bool {
        for name := range req.URL.Query() {
+               if name == "acl" && req.Method == http.MethodGet {
+                       return false
+               }
                if notimplementedObjectResourceNames[name] {
                        return true
                }
```

**Review by krishnasrinivas — APPROVED** [review]

File: cmd/generic-handlers.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/generic-handlers.go]: Yeah let me add it there, I only wanted to implement this for buckets, not objects.

**Pull request merged**